src/utils.js

```diff
@@ -89,7 +89,8 @@ export function roundToPrecision(numStr: string, scale: number, fixedDecimalScal
 
   const shoudHaveDecimalSeparator = numStr.indexOf('.') !== -1 && scale;
   const {beforeDecimal, afterDecimal, hasNagation} = splitDecimal(numStr);
-  const roundedDecimalParts = parseFloat(`0.${afterDecimal || '0'}`).toFixed(scale).split('.');
+  const floatValue = afterDecimal.length <= scale ? parseFloat(`0.${afterDecimal || '0'}`).toString() : parseFloat(`0.${afterDecimal || '0'}`).toFixed((scale));
+  const roundedDecimalParts = floatValue.split('.');
```

Comment:
Too many things happening at ternary. This could be made more readable by breaking it.

```suggestion
  const floatValue = parseFloat(`0.${afterDecimal || '0'}`);
  const floatValueStr = afterDecimal.length <= scale ? floatValue.toString() : floatValue.toFixed(scale);
  const roundedDecimalParts = floatValueStr.split('.');
```